subprojects/docs/src/docs/release/notes.md

```diff
@@ -30,6 +30,11 @@ in the next major Gradle version (Gradle 5.0). See the User guide section on the
 
 The following are the newly deprecated items in this Gradle release. If you have concerns about a deprecation, please raise it via the [Gradle Forums](https://discuss.gradle.org).
 
+### Deprecation of `--no-rebuild`
```

Comment:
:o: I wouldn't necessarily make the title specific to the long command line option as `-a` does the same thing.